### PR TITLE
SMTP STARTTLS code snippets for C#, Node.js, Python, PHP, and Perl

### PR DESCRIPTION
**Pull request by cjarnold**
Tagging @aydrian and @richleland for review.

### CODE REVIEW

File: snippets/SMTP/smtp_example.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// This example utilizes the 'nodemailer' and 'nodemailer-smtp-transport'
+// modules which can be installed using the following commands:
+// npm install nodemailer
+// npm install nodemailer-smtp-transport
+
```
Comment (richleland) [inline snippets/SMTP/smtp_example.js]: Maybe also put a note in about how a user could opt for https://github.com/SparkPost/nodemailer-sparkpost-transport as well?

File: snippets/SMTP/smtp_example.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import smtplib
+ 
+FROMADDR = 'testing@sparkpostbox.com'
+TOADDRS  = ['developers@sparkpost.com']
+SUBJECT  = 'Testing SparkPost SMTP from Python!'
+
```
Comment (richleland) [inline snippets/SMTP/smtp_example.py]: remove extra spaces before = sign per PEP8

File: snippets/SMTP/smtp_example.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
+using System;
+using System.Net;
+using System.Net.Mail;
+ 
+public class EmailTest
+{
+   public static void Main()
+   {
+        MailAddress to = new MailAddress("developers@sparkpost.com");
```
Comment (richleland) [inline snippets/SMTP/smtp_example.cs]: let's not have them send to developers@sparkpost.com - the cURL examples use `<REPLACE_WITH_YOUR_EMAIL_ADDRESS>`

File: snippets/SMTP/smtp_example.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import smtplib
+ 
+FROMADDR = 'testing@sparkpostbox.com'
+TOADDRS  = ['developers@sparkpost.com']
+SUBJECT  = 'Testing SparkPost SMTP from Python!'
+ 
+msg = ("From: %s\r\nTo: %s\r\nSubject: %s\r\n\r\nHello World!"
+       % (FROMADDR, ", ".join(TOADDRS), SUBJECT) )
```
Comment (richleland) [inline snippets/SMTP/smtp_example.py]: remove whitespace between last two closing parens per PEP8

File: snippets/SMTP/smtp_example.cs (hunk 3 above)
Comment (cjarnold) [inline snippets/SMTP/smtp_example.cs]: REPLACE_WITH_YOUR_EMAIL_ADDRESS is a bit verbose in my opinion.  I'm thinking to switch this to simply 'recipient@example.com', which seems common in other email examples.  I'm also going to add comments about sparkpostbox.com

**Pull request merged**